### PR TITLE
Part 5 refactor

### PR DESCRIPTION
**Pull request by SamanAtashi**
# OOP school library: refactor your code
Features introduced are as follow:

- Describe the SOLID principles of OOP.
- Refactor your own Ruby code after gaining new knowledge.
- Apply Ruby best practices and language style guides in code.

### CODE REVIEW

File: bin/methods/create_a_book.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def creates_a_book
+    clear
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    new_book = ActualBook.new(title, author)
+    @books << new_book
+    added_msg('Book')
+  end
```
Comment (mikethreels) [inline bin/methods/create_a_book.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed in the console, information is gathered and it is processed, this is not in line with the single responsibility principle, try to split this into smaller methods with a single responsibility. for more details have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: bin/methods/create_a_rental.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      puts "  #{idx}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_id = gets.chomp.to_i
+    puts
+    print 'Date: '
+    date = gets.chomp.to_s
+    new_rental = Rental.new(date, @books[book_id], @people[person_id])
+    @rentals << new_rental
+    added_msg('Rent')
+  end
```
Comment (mikethreels) [inline bin/methods/create_a_rental.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed in the console, information is gathered and it is processed, this is not in line with the single responsibility principle, try to split this into smaller methods with a single responsibility. for more details have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: bin/methods/create_a_student.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    new_student = Student.new(age, name, parent_permission == 'y')
+    @people << new_student
+    added_msg('Student')
+  end
```
Comment (mikethreels) [inline bin/methods/create_a_student.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed in the console, information is gathered and it is processed, this is not in line with the single responsibility principle, try to split this into smaller methods with a single responsibility. for more details have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: bin/methods/create_a_teacher.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    new_teacher = Teacher.new(specialization, age, name)
+    @people << new_teacher
+    added_msg('Teacher')
+  end
```
Comment (mikethreels) [inline bin/methods/create_a_teacher.rb]: - [x] good job cleaning up the main.rb file 👍 in this method text is printed in the console, information is gathered and it is processed, this is not in line with the single responsibility principle, try to split this into smaller methods with a single responsibility. for more details have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @SamanAtashi,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

### **optional**

- Please remove the disabled rubocop linters and try to solve the issue.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**